### PR TITLE
move basic auth state to appStore

### LINKED ISSUE #3354: redirect to basic auth hangs forever
`https://www.google.com/url?hl=en&q=http://highgrowthhandbook.rangeprojects.com/&source=gmail&ust=1472080680463000&usg=AFQjCNEoxlcS0F_6V6913poBHqRZ2lu0yw` hangs indefinitely

### PR DESCRIPTION
**Pull request by bridiver**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

fixes #3354

### CODE REVIEW

File: app/common/state/immutableUtil.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Immutable = require('immutable')
+
+const immutableUtils = {
+  makeImmutable: (obj) => {
```
Comment (bsclifton) [inline app/common/state/immutableUtil.js]: Definitely like this :+1:

File: app/browser/basicAuth.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      let username = detail.get('username')
+      let password = detail.get('password')
+      cb(username, password)
+    } else {
+      cb()
+    }
+  }
+}
+
+const doAction = (action) => {
```
Comment (bridiver) [inline app/browser/basicAuth.js]: not sure how I feel about this. I wanted to separate the electron code from the appState to make it more unit testable, but not sure if this is the best way to handle it or not. I also considered some kind of change listener, but I think that might need a larger change to make sure the notifications are efficient

File: app/sessionStore.js [hunk 3]
```diff
@@ -274,6 +275,11 @@ module.exports.cleanAppData = (data, isShutdown) => {
       })
     }
   }
+  // all data in tabs is transient while we make the transition from window to app state
+  delete data.tabs
+  // if (data.tabs) {
```
Comment (bridiver) [inline app/sessionStore.js]: I left this in here so it was clear how it would work. I wanted to keep all the state management for tabs in on place

**Comment by bbondy** [comment]
++ thanks

**Pull request merged**